### PR TITLE
User Profile

### PR DESCRIPTION
**Pull request by loicteixeira**
User is redirected to his profile upon signing in.

#### User Stories
- As a user, I want to see my profile page

#### Required Tests
- Sign into Blocitoff, are you redirect to your user show page?
- Does your user show page have the appropriate user details?

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,6 +3,15 @@ class ApplicationController < ActionController::Base
...
   protect_from_forgery with: :exception
 
+  def after_sign_in_path_for(resource)
+    sign_in_url = new_user_session_url
+    if request.referer == sign_in_url
+      me_path
+    else
+      super
+    end
+  end
```
Comment (eliotsykes) [inline app/controllers/application_controller.rb]: Hey Loïc - what's the situation the `else super` gets used for?

**Comment by eliotsykes** [comment]
Very thoroughly tested,  :+1:

**Comment by loicteixeira** [comment]
Thanks. I'm still struggling deciding where to test what (i.e. within a feature, controller, view or model spec) though.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (loicteixeira) [inline app/controllers/application_controller.rb]: Surprisingly enough, it is not only called after successful sign in, but also after reseting the password! Thanks to my other test, I was able to spot the issue immediately :)

Comment (eliotsykes) [inline app/controllers/application_controller.rb]: Thanks for explaining, I didn't realize that'd happen. I'm not sure how reliable `request.referer` is as a check - say if there were query string params or an extra slash in the referer then the `request.referer == sign_in_url` check may not always return true when intended. 

It was a slightly different scenario but I've had success customizing this behaviour by checking the controller class and the action in `after_sign_in_path_for(resource)`, e.g.
```ruby
self.is_a?(Devise::RegistrationsController) && params[:action] == 'create'
```

**Comment by eliotsykes** [comment]
> Thanks. I'm still struggling deciding where to test what (i.e. within a feature, controller, view or model spec) though.

Let's discuss in next call.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (loicteixeira) [inline app/controllers/application_controller.rb]: Agreed and updated in 3eb1dff.

**Pull request merged**